[chiayunrong] iP

cyr123456789 opened the pull request:
### Heading

- Bullet list 1
- Bullet list 2

1. Numbered list 1
2. Numbered list 2

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("code block with syntax highlighting");
    }
}
```

- [X] task list 1
- [ ] task list 2

👍 

> block
> quote

[hyperlink](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

`System.out.println("inline code");`

_text_ **formatting** ~~here~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Rename chatbot and implement initial skeletal version

Commit:
Implement echoing commands entered by user

Commit:
Add ability to store and display text

Commit:
Add ability to mark tasks as done and undone

Commit:
Add support for tracking todo, deadline and event

Commit:
Add regression tests

Commit:
Handle errors

Commit:
Add support for deleting tasks

Commit:
Fix parser bug

Commit:
Refactor tags and marks into enums

Commit:
Add support for saving and loading tasks

Commit:
Merge branch 'branch-Level-7'

Commit:
Add support for datetime objects

Commit:
Merge branch 'branch-Level-8'

Commit:
Use more OOP throughout codebase

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Merge branch 'add-gradle-support' into A-Gradle

Commit:
Add gradle support

Commit:
Merge branch 'branch-A-Gradle'

Commit:
Add minimal unit tests

Commit:
Merge branch 'branch-A-JUnit'

Commit:
Package app as JAR using gradle

Commit:
Merge branch 'branch-A-Jar'

Commit:
Add checkstyle to codebase

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Add JavaDoc comments

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Support finding tasks using a search string

Commit:
Merge branch 'branch-Level-9'

Commit:
Add GUI to the chatbot

Commit:
Merge branch 'branch-Level-10'

chew01 commented on src/main/java/Alpha.java:
```diff
@@ -0,0 +1,95 @@
+
+/**
+ * The type Alpha.
+ */
```
Perhaps consider making this class comment more descriptive?

chew01 commented on src/main/java/Processor.java:
```diff
@@ -0,0 +1,49 @@
+
+/**
+ * The type Processor.
+ */
```
Perhaps consider making this class comment more descriptive?

chew01 commented on src/main/java/com/alpha/commands/ByeCommand.java:
```diff
@@ -0,0 +1,41 @@
+     * @param taskList   Task list of the application.
+     * @param ui      Ui of the application.
+     * @param storage Storage functionality of the application.
+     */
```
Perhaps consider making this method comment more descriptive? What exactly does it do?

chew01 commented on src/main/java/com/alpha/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,38 @@
+        super(args);
+    }
+
+    /**
+     * Executes the commands.
+     *
+     * @param taskList   Task list of the application.
+     * @param ui      Ui of the application.
+     * @param storage Storage functionality of the application.
+     */
```
Perhaps consider making this method comment more descriptive?

chew01 reviewed (commented):
The code looks good 👍 
Just a few nits, particularly, the class and method comments for command types could be more descriptive.

Commit:
Refactor codebase

Improve javadocs across entire codebase.

Avoid magic literals and unused variables to reduce confusion.

Make code more obvious to readers.

Structure code logically.

Commit:
Merge pull request #1 from chiayunrong/branch-A-CodeQuality

Improve code quality

Commit:
Add assertions

Commit:
Merge pull request #2 from chiayunrong/branch-A-Assertions

Add assertions

Commit:
Add support for GHA CI

Commit:
Merge pull request #3 from chiayunrong/A-CI

Add support for Continuous Integration

Commit:
Add support for detecting duplicate tasks

Commit:
Merge pull request #4 from chiayunrong/branch-C-DetectDuplicates

Add support for detecting duplicate tasks

erohsikivar commented on src/main/java/Alpha.java:
```diff
@@ -0,0 +1,151 @@
+/**
+ * The Alpha class.
+ */
+public class Alpha extends Application {
```
You did a great job using different methods to have a better GUI.

erohsikivar commented on src/main/java/com/alpha/commands/FindCommand.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Override
+    public String execute() {
```
Good use of String Builder.

erohsikivar commented on src/main/java/com/alpha/commands/ToDoCommand.java:
```diff
@@ -0,0 +1,37 @@
+     * @param name    The name of the task.
+     * @param taskList The task list.
+     */
+    public ToDoCommand(String name, TaskList taskList) throws DuplicateTaskException {
```
I never thought of handling duplicate tasks. Good job

erohsikivar reviewed (commented):
Overall LGTM, good job.

Commit:
Add user guide and screenshot

Commit:
Merge pull request #6 from chiayunrong/branch-A-UserGuide

Add user guide and screenshot

Commit:
Minor fixes to build.gradle and javadocs
